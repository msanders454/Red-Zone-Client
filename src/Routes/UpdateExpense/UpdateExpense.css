.EditExpense{
    color: white;
}

a {
    color: hotpink;
  }