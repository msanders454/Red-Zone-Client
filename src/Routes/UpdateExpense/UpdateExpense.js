/* eslint-disable react/prop-types */
import React, { Component } from 'react';
import RedZoneContext from '../../RedZoneContext';
import config from '../../config';
import ExpenseForm from '../../ExpenseForm/ExpenseForm';

/*
* Used to patch user expense logs.
*/ 
export default class UpdateExpense extends Component {
    static contextType = RedZoneContext;

    state = {
        error: null,
        id: null,
        date: null,
        amount: null,
        style: null,
        description: null,
    };
    componentDidMount() {
        const { expenseId } = this.props.match.params;
        fetch(`https://serene-ridge-50508.herokuapp.com/api/expenses/${expenseId}`, {
            method: "GET",
            headers: {
                "content-type": "application/json",
            }
        })
            .then(res => {
                if (!res.ok) return res.json().then(error => Promise.reject(error));

                return res.json();
            })
            .then(responseData => {
                this.setState({
                    id: responseData.id,
                    date: responseData.date,
                    amount: responseData.amount,
                    style: responseData.style,
                    description: responseData.description,
                    usernumber: responseData.usernumber
                });
            })
            .catch(error => {
                this.setState({ error });
            });
    }
    handleSubmit = (expense, callback) => {
        this.setState({ error: null })
        const { expenseId } = this.props.match.params;
        fetch(config.API_ENDPOINT + `/${expenseId}`, {
            method: 'PATCH',
            body: JSON.stringify(expense),
            headers: {
                'content-type': 'application/json',
            },
        })
            .then(res => {
                if (!res.ok)
                return res.json().then(error => Promise.reject(error))
            })
            .then(() => {
                callback(callback)
                this.context.updateExpense(expense)
                this.props.history.push('/expenses')
            })
            .catch(error => {
                console.error(error)
                this.setState({ error })
            })
    }
    handleClickCancel = () => {
        this.props.history.push('/expenses')
    };

    render() {
        const { error, id, date, amount, style, description, usernumber } = this.state;
        const expense = { id, date, amount, style, description, usernumber}
        return (
            <section className='EditExpense'>
                {id && (
                    <ExpenseForm
                        onSubmit={this.handleSubmit}
                        onCancel={this.handleClickCancel}
                        error={error}
                        expense={expense}
                    />
                )}
            </section>
        );
    }
}