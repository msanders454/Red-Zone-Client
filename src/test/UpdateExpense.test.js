/* eslint-disable no-undef */
import React from "react";
import { shallow, configure } from "enzyme";
import UpdateExpense from "../Routes/UpdateExpense/UpdateExpense";
import Adapter from "enzyme-adapter-react-16";
 
configure({ adapter: new Adapter() });
test("should render UpdateExpense correctly", () => {
  const match = {
        params : {expenseId : 1}
        }
  const wrapper = shallow(<UpdateExpense match={match}/>);
  expect(wrapper).toMatchSnapshot();
});