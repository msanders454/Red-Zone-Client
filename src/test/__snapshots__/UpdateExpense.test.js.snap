// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render UpdateExpense correctly 1`] = `ShallowWrapper {}`;
